plugins/node/instrumentation-amqplib/src/amqplib.ts

```diff
@@ -409,9 +410,18 @@ export class AmqplibInstrumentation extends InstrumentationBase {
         const headers = msg.properties.headers ?? {};
         const parentContext = propagation.extract(ROOT_CONTEXT, headers);
         const exchange = msg.fields?.exchange;
-        const span = self.tracer.startSpan(
-          `${queue} process`,
-          {
+        let span: Span;
+        if (self._config.useLinksForConsume) {
```

Comment:
the 2 cases in the if-else contains some duplication.

WDYT about preparing the `links` array only once (with 0 or 1 elements) and the `parent` based on the instrumentation configuration, then just invoking `self.tracer.startSpan` once with these values? or at least calculate the `attributes` once, outside the if-else

Reply:
Would a null parent be fine in this instance?  Because with the config option, the parentcontext should not exists.  If that's fine, then I can make those changes.

Reply:
`undefined` is fine. It's the default anyway

Reply:
I'd do something like this

```typescript
        const links: Link[] = [];
        if (self._config.useLinksForConsume) {
          const parentSpanContext = trace.getSpan(parentContext)?.spanContext();
          if (parentSpanContext) {
            links.push({ context: parentSpanContext });
          }
        }

        const span = self.tracer.startSpan(`${queue} process`, {
          kind: SpanKind.CONSUMER,
          attributes: {
            ...channel?.connection?.[CONNECTION_ATTRIBUTES],
            [SemanticAttributes.MESSAGING_DESTINATION]: exchange,
            [SemanticAttributes.MESSAGING_DESTINATION_KIND]:
              MessagingDestinationKindValues.TOPIC,
            [SemanticAttributes.MESSAGING_RABBITMQ_ROUTING_KEY]:
              msg.fields?.routingKey,
            [SemanticAttributes.MESSAGING_OPERATION]:
              MessagingOperationValues.PROCESS,
            [SemanticAttributes.MESSAGING_MESSAGE_ID]:
              msg?.properties.messageId,
            [SemanticAttributes.MESSAGING_CONVERSATION_ID]:
              msg?.properties.correlationId,
          },
          links,
        },
        self._config.useLinksForConsume ? undefined : parentContext,
        );
```

Reply:
I had to change it a little as parentContext was used deeper in the code so I wanted to explicitly make it undefined in this instance so the trace was not continued.